### PR TITLE
Add exception handling when creating /var/lib/waagent/events folder

### PR DESCRIPTION
**Pull request by pgombar**
## Description

Make telemetry exceptions non-blocking for the agent. If we are unable to create the `events` folder for whatever reason, log the error periodically, but continue running the daemon and extension handler to support extensions workflows.

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1641)

### CODE REVIEW

File: azurelinuxagent/common/event.py [hunk 1]
```diff
@@ -208,7 +208,11 @@ def save_event(self, data):
             logger.warn("Cannot save event -- Event reporter is not initialized.")
             return
 
-        fileutil.mkdir(self.event_dir, mode=0o700)
+        try:
+            fileutil.mkdir(self.event_dir, mode=0o700)
+        except (IOError, OSError) as e:
+            msg = "Failed to create events folder {0}. Error: {1}".format(self.event_dir, ustr(e))
+            raise EventError(msg)
```
Comment (pgombar) [inline azurelinuxagent/common/event.py]: The caller of this method handles `EventError` exceptions.

Comment (narrieta) [reply azurelinuxagent/common/event.py]: In the caller
```
        try:
            self.save_event(json.dumps(data))
        except EventError as e:
            logger.error("{0}", e)
```

can you make the log statement periodic? thanks

Comment (narrieta) [reply azurelinuxagent/common/event.py]: (intention is not to flood the serial console)

Comment (pgombar) [reply azurelinuxagent/common/event.py]: Makes sense, updated to every 15 minutes.

File: azurelinuxagent/common/event.py [hunk 2]
```diff
@@ -290,7 +290,7 @@ def _add_event(self, duration, evt_type, is_internal, is_success, message, name,
         try:
             self.save_event(json.dumps(data))
         except EventError as e:
-            logger.error("{0}", e)
+            logger.periodic_error(logger.EVERY_FIFTEEN_MINUTES, ustr(e))
```
Comment (narrieta) [inline azurelinuxagent/common/event.py]: We agreed we would prepend "[PERIODIC]" to the message of periodic events/logs to help during debugging

Comment (pgombar) [reply azurelinuxagent/common/event.py]: Thanks for the reminder, updated.

**Pull request merged**